swesmith/bug_gen/adapters/typescript.py

```diff
@@ -0,0 +1,275 @@
+                    ">=",
+                ]:
+                    self._tags.add(CodeProperty.HAS_OFF_BY_ONE)
+
+    @property
+    def name(self) -> str:
+        return self._extract_name_from_node()
+
+    def _extract_name_from_node(self) -> str:
+        # Function declarations
+        if self.node.type == "function_declaration":
+            return self._find_child_text("identifier")
+        # Method definitions
+        if self.node.type == "method_definition":
+            return self._find_child_text("property_identifier")
+        # Class declarations
+        if self.node.type == "class_declaration":
+            return self._find_child_text("identifier")
+        # Variable declarations with function expressions
+        if self.node.type == "variable_declarator":
+            return self._find_child_text("identifier")
+        # Assignment expressions with function expressions
+        if self.node.type == "assignment_expression":
+            return self._find_child_text("identifier")
+        return ""
+
+    def _find_child_text(self, child_type: str) -> str:
+        for child in self.node.children:
+            if child.type == child_type:
+                return child.text.decode("utf-8")
+        return ""
+
+    @property
+    def signature(self) -> str:
+        for child in self.node.children:
+            if child.type in ["statement_block", "class_body"]:
+                body_start_byte = child.start_byte - self.node.start_byte
+                signature = self.src_code[:body_start_byte].strip()
+                if signature.endswith(" {"):
+                    signature = signature[:-2].strip()
+                return signature
+        if self.node.type == "arrow_function" and "=>" in self.src_code:
+            return self.src_code.split("=>")[0].strip() + " =>"
+        if self.node.type == "variable_declarator":
+            src_lines = self.src_code.split("\n")
+            first_line = src_lines[0]
+            if " = function" in first_line:
+                brace_pos = first_line.find(" {")
+                if brace_pos != -1:
+                    return first_line[:brace_pos].strip()
+                else:
+                    result = first_line.strip()
+                    if result.endswith(";"):
+                        result = result[:-1].strip()
+                    return result
+        return self.src_code.split("\n")[0].strip()
+
+    @property
+    def stub(self) -> str:
+        signature = self.signature
+        if self.node.type == "class_declaration":
+            return f"{signature} {{\n\t// {TODO_REWRITE}\n}}"
+        elif self.node.type == "arrow_function":
+            if "=>" in signature:
+                return f"{signature} {{\n\t// {TODO_REWRITE}\n}}"
+            else:
+                return f"{signature} => {{\n\t// {TODO_REWRITE}\n}}"
+        else:
+            return f"{signature} {{\n\t// {TODO_REWRITE}\n}}"
+
+    @property
+    def complexity(self) -> int:
```

Comment:
I think we only have a minimal test for complexity at the moment? Could we improve this?

Reply:
Oh yeah that complexity test is terrible, I'll fix it. This is definitely not great:
```
for e in entities:
        assert e.complexity >= 1
```